debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -775,9 +775,7 @@ public void readSchema(Tables tables, String databaseCatalog, String schemaNameP
                             // ignore, some drivers don't have this index - e.g. Postgres
                         }
                         column.generated("YES".equalsIgnoreCase(autogenerated));
-                        if ( column.jdbcType() == Types.ARRAY) {
```

Comment:
Why is it that you removed this check? It actually causes several test failures due to the assertion that makes sure that `componentType()` only is invoked for an array-typed column. I'll re-add it.